### ISSUE TITLE
Add support for Cloud Bigtable replication

### ISSUE THREAD

**Issue by mbrukman** [terraform-providers/terraform-provider-google#1672]
Currently, support for Cloud Bigtable assumes a 1:1 correspondence between instances and clusters, which was true up until recently, when we launched support for replication (see [blog post](https://cloudplatform.googleblog.com/2018/05/regional-replication-for-cloud-bigtable.html)).

In [the docs](https://github.com/terraform-providers/terraform-provider-google/blob/7e64b3f1c1124f612f784800f0a168a47ca877c5/website/docs/r/bigtable_instance.html.markdown), an instance and a cluster are configured together, and assumes that there is exactly 1 of each `instance` and `cluster` created together:

```hcl
resource "google_bigtable_instance" "instance" {
  name         = "tf-instance"
  cluster_id   = "tf-instance-cluster"
  zone         = "us-central1-b"
  num_nodes    = 3
  storage_type = "HDD"
}
```

What we need to do is enable configuring an instance (an abstract container) together with a list of clusters, each of which has a unique cluster id (so it could actually be a map as well).

In fact, the list of clusters is how it's implemented in the [`CreateInstanceRequest` proto](https://github.com/googleapis/googleapis/blob/master/google/bigtable/admin/v2/bigtable_instance_admin.proto#L187-L207) (note: ignore the comment about "exactly 1 cluster" per instance; that will be fixed soon).

**Comment by emilymye** [comment]
To document, this requires: 
- Updating bigtable go lib to v2
- Changing `google_bigtable_instance` schema to have cluster schema block (possibly a breaking change)

**Comment by rackonnoiter** [comment]
What is the recommended method of updating the bigtable go library? Manually copy it over from [the repo](https://github.com/GoogleCloudPlatform/google-cloud-go/tree/master/bigtable) into the vendor directory?

**Comment by paddycarver** [comment]
@mbrukman can you confirm that the note in the [API docs](https://cloud.google.com/bigtable/docs/reference/admin/rest/v2/projects.instances/create) for creating an instance about `clusters` ("Currently exactly one cluster must be specified.") is incorrect? If so, does the API request not care about the number of clusters at all, or does it need 1 _or more_? And is there any difference between creating the clusters with the instance and creating the clusters using the [clusters API](https://cloud.google.com/bigtable/docs/reference/admin/rest/v2/projects.instances.clusters/create)?

I'd love any clarification so we can get this design right.

**Comment by mbrukman** [comment]
@paddycarver wrote:

> @mbrukman can you confirm that the note in the API docs for creating an instance about clusters ("Currently exactly one cluster must be specified.") is incorrect? If so, does the API request not care about the number of clusters at all, or does it need 1 or more?

Sorry, that documentation page is out of date and needs to be re-generated. The [source proto][proto] has the correct text: _"Currently, at most two clusters can be specified."_

> And is there any difference between creating the clusters with the instance and creating the clusters using the clusters API?

No, the resulting clusters are the same. So you can either:

* create an instance with a single cluster and add a cluster afterwards
* create an instance from scratch with two clusters

@garye, @dmmcerlean — please correct me if I'm wrong.

[proto]: https://github.com/googleapis/googleapis/blob/79ab27f3b70ebc221e265d2e8ab30a8cc2d21fa2/google/bigtable/admin/v2/bigtable_instance_admin.proto#L206

**Comment by garye** [comment]
The resulting clusters are the same regardless of creating one at a time or both together, but the behavior of the default application profile is different. If both clusters are created at once, the profile will be set up for _multi-cluster routing_. Otherwise, it will be set up for _single-cluster routing_ to the _original cluster_ that was created via `CreateInstance`.

**Comment by paddycarver** [comment]
This was resolved in #2161, I believe.

**Comment by rileykarson** [comment]
I think we have the right shape for replication now, but didn't implement it yet because we were only making the breaking change for `1.19`.
